### DISCUSSION TITLE
additional redirect urls not working

### DISCUSSION THREAD

**Discussion by oouazize** [supabase/supabase#27250] [Questions]
I have a **self hosted Supabase** using docker, and the **additional redirect URLs** didn't trigger when this code ran
const { data, error } = await supabase.auth.resetPasswordForEmail(email.value, {
redirectTo: `${config.public.FRONTEND_DOMAIN}/recover`,
});

it's always redirects to SITE_URL even the config.public.FRONTEND_DOMAIN contains one of the URLs I set in the ADDITIONAL_REDIRECT_URLS

here is how I set it 

```
ADDITIONAL_REDIRECT_URLS="URL1,URL2,URL3,...."
```

and here is the environment in docker compose

```
      GOTRUE_SITE_URL: ${SITE_URL}/recover
      GOTRUE_URI_ALLOW_LIST: ${ADDITIONAL_REDIRECT_URLS}
```

It's either a bug or something I missing.

**Comment by GaryAustin1** [comment]
I don't use self hosting but do you have no spaces between the URLs and are they the full urls including http?

**Reply by oouazize** [reply]
check this
https://stackoverflow.com/questions/75418419/supabase-auth-redirectto-not-working-for-oauth

I also tried to put only one URL but it doesn't work

**Comment by Shooteger** [comment]
This really seems to be a bug. We try to access this from our Nuxt 3 Frontend with nuxt-supabase package, but is does not work. The redirectTo parameter seems always to be ignored, no matter what settings are done in supabase. It only accepts the SITE_URL, not any other which we send.

Currently we try to find a workaround on our client side but this should be investigated.

To reproduce just create a frontend (we use Nuxt 3 with nuxt-supabase package) and supabase instance (self host) and use "supabase.auth.resetPasswordForEmail()" with redirectTo parameter.

We have the additional redirect URLS correctly as in the question here without spaces:

`ADDITIONAL_REDIRECT_URLS="URL1,URL2,URL3,...."`

You can make redirectTo to a fixed string too, same result, it gets ignored by supabase and yes, our FRONTEND_DOMAIN has a valid value, we console.log it on localhost, it gets correctly parsed into redirectTo.

```
const { data, error } = await supabase.auth.resetPasswordForEmail(email.value, {
            redirectTo: `${config.public.FRONTEND_DOMAIN}/recover`,
 });
```

**Comment by hitaspdotnet** [comment]
This is a bug. We try to access supabase auth from our deployed nextjs and localhost:
SITE_URL=https://w.xyz
SERVICE_FQDN_SUPABASEKONG=https://api.w.xyz
ADDITIONAL_REDIRECT_URLS=http://localhost:3000

but always redirects to https://w.xyz as anon (session is empty)

**Reply by junkaiman** [reply]
same here. looking for an answer

**Reply by sapphire-haeward** [reply]
I too am also in search on answers and I'm facing the same issue.

**Reply by viworks-niros** [reply]
still not a fix? is this related to SDK v2?

**Reply by djsisson** [reply]
it works fine

it needs to exactly match the redirect, so any stray ```/``` would not match, so make sure you use glob pattern matching since your redirect url would not be exactly ```http://localhost:3000 ```

so anything like

```http://localhost:3000/**``` 

this would match any route since ```*``` by itself would only match top level routes

see here

https://github.com/supabase/auth?tab=readme-ov-file#top-level

URI_ALLOW_LIST env section

**Reply by noobSam1122** [reply]
I am facing the issue right now, I have set up the redirect URLs as follow

https://192.168.0.241:8000/

https://192.168.0.241:8000/**

and use

```
        const { data, error } = await supabase.auth.signInWithOAuth({
            provider: 'google',
            options: {
                redirectTo: window.location.origin + "/user",
            }
        })
```
It always redirects to my SITE_URL

**Reply by djsisson** [reply]
Was that a typo or did you put https? Since it's a private ip I would assume it should be http you are trying to redirect to, which wouldn't match

**Reply by DoiiarX** [reply]
me too

**Comment by brondibur** [comment]
This is a problem during local development and testing. I've created an issue for this: https://github.com/supabase/supabase/issues/36640

**Comment by hitaspdotnet** [comment]
I found a workaround for this by setting:

```yaml
- 'GOTRUE_SITE_URL=${SITE_URL:-http://localhost:3000}'
- 'GOTRUE_URI_ALLOW_LIST=${ADDITIONAL_REDIRECT_URLS}'
```

then:
```env
ADDITIONAL_REDIRECT_URLS=https://*.your-domain.com/**
SITE_URL=
```

leave site url empty at local testing. after tests `https://your-domain.com`
